socranop/data/man/socranop-session-service.1

```diff
@@ -0,0 +1,132 @@
+.\"
+.\" ======================================================================
+.\"
+.TH SOCRANOP\-SESSION\-SERVICE 1 "2021\-07\-04" "${PACKAGE} ${VERSION}" "User commands"
+.\"
+.\" ======================================================================
+.\"
+.SH NAME
+socranop\-session\-service \- session D\-Bus service for socranop
+.\"
+.\" ======================================================================
+.\"
+.\" usage: socranop-session-service [-h] [--version]
+.\"
+.\" The socranop D-Bus service.
+.\"
+.\" optional arguments:
+.\"   -h, --help     show this help message and exit
+.\"   --version      show program's version number and exit
+.\"   -v, --verbose  Enable more verbose output, largely for debugging
+.\"
+.SH SYNOPSIS
+.B "socranop\-session\-service"
+.RB (\| \-\-help \|| \-\-version \|)
+.br
+.B "socranop\-session\-service"
+.RB [\| \-\-verbose \|]
+.\"
+.\" ======================================================================
+.\"
+.SH DESCRIPTION
+.PP
+The
+.B "socranop\-session\-service"
+D\-Bus service is bus activated, so it does not need to be run
+explicity in normal operation.
+.PP
+The D\-Bus service fulfils two main functions:
+.IP "stand in for the mixer state" 8
+As the Notepad series of mixers does not appear to have a USB command to query the state which can be set with a USB command, the D\-Bus service remembers the hardware state to provide a halfway sane behaviour.
+.IP "notify other processes of mixer state changes"
+If you have e.g. a
+.B "socranop-gui"
+window open, and change the mixer state from the command line with
+.BR "socranop-ctl \-\-set" ,
+the
+.B "socranop-gui"
+window will show that state change immediately.
+.\"
+.\" ======================================================================
+.\"
+.SH OPTIONS
+.PP
+If no options are given on the command line,
+.B socranop\-session\-service
+will just start as a D\-Bus service and keep running until a D\-Bus
+client requests it to shut down.
+.TP
+.BR \-\-help
+Print command line help and exit.
+.TP
+.BR \-\-version
+Print the program version and exit.
+.TP
+.RB (\| \-v | \-\-verbose \|)
+Make the program output more information. Helpful for debugging.
+.\"
+.\" ======================================================================
+.\"
+.SH EXIT STATUS
+If successful, returns 0. Non-0 exit codes indicate an error.
+.\"
+.\" ======================================================================
+.\"
+.SH ENVIRONMENT
+.TP
+.B XDG_CONFIG_HOME
+See the state file description in the FILES section below.
+.\"
+.\" ======================================================================
+.\"
+.SH FILES
+.TP
+.B ~/.config/${PACKAGE}/state/*.state
+The state file is used to keep track of the state the device has been set to.
+.IP
+If the \fBXDG_CONFIG_HOME\fR environment variable is set, use its value instead of the \fB~/.config\fR part.
+.\"
+.\" ======================================================================
+.\"
+.\" .SH NOTES
```

Comment:
Somewhere in this file it's probably worth noting that in order for this service to interact with the USB device, user device access needs to be given, with a reference to the `socranop-installtool` that can help set that up.

Reply:
I would prefer a reference to a separate document which explains the permission stuff, either to a `PERMISSIONS.md` text file or to a `socranop-permissions(7)` man page.

In https://github.com/spkr-beep/beep, I have five `*.md` documentation files in addition to the `beep(1)` man page:

  * README.md: General overview of what beep does, what it does not do, and where to find which kind of more detailed information.
  * INSTALL.md: How to build and install the software.
  * PERMISSIONS.md: How to set up the device permissions.
  * DEVELOPMENT.md: How to hack on the software, coding standards, testing, APIs used, architecture, release checklist, etc.
  * PACKAGING.md: How to create distribution packages (RPM, DEB, etc.)

This allows me to relatively easily refer to PERMISSIONS.md whenever the topic of permissions comes up in one of the other files (e.g. in both INSTALL.md and PACKAGING.md).

Something similar might be a good idea for socranop as well.

While one could argue that the information in `PERMISSIONS.md` should really be in a `socranop-permissions(7)` man page so that linking there from the other man pages is trivial, the information in `PERMISSIONS.md` should also be available before any man pages are installed, and also displayy nicely on https://github.com/socratools/socranop/, which means definitively not a man page.

Reply:
I like that idea.  Directing someone from the manpage to the github space to explain a one-time setup procedure isn't bad.